include/zephyr/drivers/adc.h

```diff
@@ -768,6 +768,7 @@ static inline int adc_channel_setup_dt(const struct adc_dt_spec *spec)
  *                  interval was too small. All requested samples are written
  *                  in the buffer, but at least some of them were taken with
  *                  an extra delay compared to what was scheduled.
+ * @retval <0       Any other negative value indicates an error condition.
```

Comment:
@anangl please confirm this makes sense for you

Reply:
Yes, it's fine by me.